src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -254,11 +318,27 @@ async def _finish_handler_with_unary_response(RPCState rpc_state,
     stream-unary handlers.
     """
     # Executes application logic
-    
-    cdef object response_message = await unary_handler(
-        request,
-        servicer_context,
-    )
+    cdef object response_message
+    cdef _SyncServicerContext sync_servicer_context
+
+    if _is_async_handler(unary_handler):
```

Comment:
Is it possible to make this conditional a preprocessing step? I suspect that would give us a performance boost.

Reply:
In the interceptor PR, mmx mentioned we can register handlers to methods in Core (see [Core API](https://github.com/grpc/grpc/blob/master/include/grpc/grpc.h#L387).

This is boosting the type check speed, but I wasn't sure if we can leverage API above. We can do it in a separate PR. Created: [Aio] Preprocess the categorization of method handlers #22824